packages/strapi-parts/src/Layout/HeaderLayout.js

```diff
@@ -1,72 +1,163 @@
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import { H1, Subtitle, P, H2 } from '../Text';
 import { Box } from '../Box';
 import { Row } from '../Row';
 
-export const HeaderLayout = ({
-  navigationAction,
-  primaryAction,
-  secondaryAction,
-  subtitle,
-  title,
-  sticky,
-  ...props
-}) => {
-  if (sticky) {
+const useElementOnScreen = (options) => {
+  const containerRef = useRef(null);
+  const [isVisible, setIsVisible] = useState(true);
+
+  const cb = ([entry]) => {
+    setIsVisible(entry.isIntersecting);
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(cb, options);
+    if (containerRef.current) observer.observe(containerRef.current);
+
+    return () => {
+      if (containerRef.current) observer.unobserve(containerRef.current);
+    };
+  }, [containerRef, options]);
+
+  return [containerRef, isVisible];
+};
+
+const useHeaderSize = () => {
+  const baseHeaderLayoutRef = useRef(null);
+  const [headerSize, setHeaderSize] = useState(null);
+
+  useEffect(() => {
+    if (baseHeaderLayoutRef.current) {
+      setHeaderSize(baseHeaderLayoutRef.current.getBoundingClientRect());
+    }
+  }, [baseHeaderLayoutRef]);
+
+  return {
+    baseHeaderLayoutRef,
+    headerSize,
+  };
+};
+
+export const HeaderLayout = (props) => {
+  const [containerRef, isVisible] = useElementOnScreen({
+    root: null,
+    rootMargin: '0px',
+    threshold: 0,
+  });
+  const { baseHeaderLayoutRef, headerSize } = useHeaderSize();
+
+  return (
+    <>
+      <Box style={{ height: headerSize?.height }} ref={containerRef}>
+        {isVisible && <BaseHeaderLayout ref={baseHeaderLayoutRef} {...props} />}
+      </Box>
+      {!isVisible && <BaseHeaderLayout {...props} sticky width={headerSize?.width} />}
+    </>
+  );
+};
+
+HeaderLayout.displayName = 'HeaderLayout';
+
+const StickyBox = styled(Box)`
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: ${(props) => props.width}px;
+`;
+
+export const BaseHeaderLayout = React.forwardRef(
+  ({ navigationAction, primaryAction, secondaryAction, subtitle, title, sticky, width, ...props }, ref) => {
+    if (sticky) {
+      return (
+        <StickyBox
```

Comment:
Since it's a forwarded ref, maybe you could pass the ref to this box also? 😊

Reply:
The observer needs to check for the static component only; not the sticky one. If needed, we could update the component to support custom refs at some point.